core/camel-support/src/main/java/org/apache/camel/support/PropertyBindingSupport.java

```diff
@@ -1438,23 +1438,46 @@ private static boolean isParameterMatchingType(Class<?> parameterType, Class<?>
         return parameterType.isAssignableFrom(expectedType);
     }
 
+    /**
+     * Resolves the value as either a class, type or bean.
+     *
+     * @param      camelContext the camel context
+     * @param      name         the name of the bean. Never used
+     * @param      value        how to resolve the bean with a prefix of either #class:, #type: or #bean:
+     * @return                  the resolve bean
+     * @throws     Exception    is thrown if error resolving the bean, or if the value is invalid.
+     *
+     * @deprecated              use {@link #resolveBean(CamelContext, Object)}
+     */
+    @Deprecated
+    public static Object resolveBean(CamelContext camelContext, String name, Object value) throws Exception {
```

Comment:
Just remove the old method as its not used anywhere outside core

Reply:
well, I was using it :)

Reply:
done